src/invidious/helpers/tokens.cr

```diff
@@ -78,32 +78,31 @@ def validate_request(token, session, request, key, db, locale = nil)
     raise translate(locale, "Hidden field \"token\" is a required field")
   end
 
-  if !Crypto::Subtle.constant_time_compare(token["signature"].to_s, sign_token(key, token))
-    raise translate(locale, "Invalid signature")
+  expire = token["expire"]?.try &.as_i
+  if expire.try &.< Time.utc.to_unix
+    raise translate(locale, "Token is expired, please try again")
   end
 
   if token["session"] != session
     raise translate(locale, "Erroneous token")
   end
 
-  if token["nonce"]? && (nonce = db.query_one?("SELECT * FROM nonces WHERE nonce = $1", token["nonce"], as: {String, Time}))
-    if nonce[1] > Time.utc
-      db.exec("UPDATE nonces SET expire = $1 WHERE nonce = $2", Time.utc(1990, 1, 1), nonce[0])
-    else
-      raise translate(locale, "Erroneous token")
-    end
+  if !Crypto::Subtle.constant_time_compare(token["signature"].to_s, sign_token(key, token))
+    raise translate(locale, "Invalid signature")
   end
 
   scopes = token["scopes"].as_a.map { |v| v.as_s }
```

Comment:
The scope check could be made _after_ verifying the signature, what do you think @omarroth?

Reply:
Makes sense to me.